### ISSUE TITLE
weird configure error

### ISSUE THREAD

**Issue by lightmare** [mapnik/mapnik#3918]
Just doing `make` or `./configure` after a few months, I get this:

```
...
Checking for requested plugins dependencies...
Checking for gdal-config --libs... SConscript:~/git/mapnik/master/SConstruct  took 3261.406 ms
UnicodeDecodeError: 'ascii' codec can't decode byte 0xa0 in position 40: ordinal not in range(128):
  File "~/git/mapnik/master/SConstruct", line 1672:
    if conf.parse_config('GDAL_CONFIG',checks='--libs'):
  File "~/git/mapnik/master/scons/scons-local-3.0.1/SCons/SConf.py", line 659:
    ret = self.test(context, *args, **kw)
  File "~/git/mapnik/master/SConstruct", line 605:
    ret = context.TryAction(cmd)[0]
  File "~/git/mapnik/master/scons/scons-local-3.0.1/SCons/SConf.py", line 817:
    return self.sconf.TryAction(*args, **kw)
  File "~/git/mapnik/master/scons/scons-local-3.0.1/SCons/SConf.py", line 612:
    outputStr = self.lastTarget.get_contents().decode()
make: *** [mapnik] Error 2
```

The offending decode is in `TryAction`, for some reason it takes the contents of `lastTarget` -- which I presume is the target built by a previous configure test unrelated to this one, and which in my case happens to be some ELF binary in .sconf_temp/ -- and tries to turn that into a unicode string?

What's going on, and how am I supposed to fix this?

**Comment by springmeyer** [comment]
wierd. I wonder if this is some kind of scons bug that would be avoided/fixed by updating to a newer version?

**Comment by artemp** [comment]
Looks very familiar, I'll take a look and fix it next week when I'm back from travels. 
/cc @lightmare @springmeyer

**Comment by lightmare** [comment]
scons master still has it. I've submitted an issue there, but it may also be a misuse of `TryAction` in our script.

simplest workaround:
`rm .sconf_temp/*`

**Comment by lightmare** [comment]
So the invalid `.decode()` has been fixed in scons. But `TryAction` is still required to produce a text file, which means we've been using it wrong all along.

One way to fix this would be to wrap all calls to `TryAction` in a function that adds output redirection `> $TARGET`, or construct a `CommandAction` with a list of arguments that'll be properly quoted -- which is what I started writing.

However, after replacing a few calls I realized that the misuse of `TryAction` might be not just failing to write an output file, but that even using `TryAction` for the purpose of checking the return status of a command may be wrong. See, all the `sconf.Try*` methods are about running some commands that produce a file under `.sconf_temp` and cache the result for future invocations.

I don't think that conftest caching is desirable for stuff like `pkg-config --exists foo`. We might be better off with just conveniently wrapped `Popen`.
